pyanaconda/modules/storage/partitioning/manual/manual_module.py

```diff
@@ -162,15 +167,65 @@ def _create_request_for_device(self, device):
         :return: an instance of MountPointRequest
         """
         request = MountPointRequest()
-        request.device_spec = device.path
+        request.device_spec = self._btrfs_device_spec(device) or device.path
```

Comment:
Would it be possible to change `MountPointRequest` here to have more information (like adding a `request.subvolspec` here)? I don't like the idea of taking Blivet device, reducing it to one string (path or uuid + subvolume name in case of btrfs) just to use `devicetree.resolve_devspec` later to get the Blivet device back. I understand that a Blivet device cannot be send over DBus, but inventing a special "device specification string" just for btrfs subvolumes seems like a very complicated solution.

Reply:
I preferred it in the beginning as well, but then I realized that if I don't want to change API of https://github.com/rvykydal/anaconda/blob/31db7b8e5f6b423ea8a3ec2663b42c0720a771e9/pyanaconda/modules/storage/devicetree/viewer_interface.py#L104 by adding kind of special argument, I need to work only with device_spec and then having another request attribute would be redundant. Moreover, we specify all the devices just by the spec so far in the request (ie reduce the device to the device specification) so it would be only consistent to use just the spec also for btrfs files.
Constructing the device_spec as and arbitrary string and single argument for anaconda's ResolveDevice (and blivet's resolve_device?) seemed more future proof than adding arguments to me.